api/workflow/model.go

```diff
@@ -119,26 +124,34 @@ type RequestStep struct {
 	Approvers []RequestStepApprover `json:"approvers,omitempty"`
 }
 
-// Request request response definition
+// Request access request definition
 type Request struct {
-	ID                   string        `json:"id,omitempty"`
-	Author               string        `json:"author,omitempty"`
-	Created              string        `json:"created,omitempty"`
-	Updated              string        `json:"updated,omitempty"`
-	UpdatedBy            string        `json:"updated_by,omitempty"`
-	Name                 string        `json:"name,omitempty"`
-	RequestJustification string        `json:"request_justification,omitempty"`
-	GrantType            string        `json:"grant_type,omitempty"`
-	GrantStart           string        `json:"grant_start,omitempty"`
-	GrantEnd             string        `json:"grant_end,omitempty"`
-	Action               string        `json:"action,omitempty"`
-	Status               string        `json:"status,omitempty"`
-	Comment              string        `json:"comment,omitempty"`
-	WorkflowID           string        `json:"workflow,omitempty"`
-	FloatingLength       int           `json:"floating_length,omitempty"`
-	TargetRoles          []string      `json:"target_roles,omitempty"`
-	Steps                []RequestStep `json:"steps,omitempty"`
-	TargetUser           User          `json:"target_user,omitempty"`
-	Requester            User          `json:"requester,omitempty"`
-	RequestedRole        Role          `json:"requested_role,omitempty"`
+	ID                        string        `json:"id"`
```

Comment:
Removing omitempty changes the current API implementation in the SDK. I suggest for v1 we do not make any breaking changes. This Access Request must closely follows the API spec of workflow-engine. For example, there is no WorkflowID in the spec (note that json '-' also means it does not get to be marshalled)

While updating this and preparing for v2 in the future, please note that the idea of using a single model for both the resource and the creation/edition/deletion actions of that resource brings a lot of troubles. This is amplified for complex resources in which there are (client-side) read-only attributes, omitempty attributes, attributes with non-zero default value...